Makefile

```diff
@@ -84,6 +90,76 @@ else
 	@echo Update complete
 endif
 
+## zkgroup
+build-zkgroup:
+	@echo "Found cargo with version $(CARGO_VERSION)"
+	@echo "Found go with version $(GO_VERSION)"
+	@echo "Found go with version $(GIT_VERSION)"
+ifeq ($(UNAME_S), Linux)
+ifeq ($(UNAME_HARDWARE_PLATTFORM), x86_64)
+	@echo "get zkgroup $(PLATFORM)"
+	go get -d github.com/nanu-c/zkgroup
+	&& git submodule update \
+	&& cd lib/zkgroup \
+	&& cargo build --release --verbose
+	mv libzkgroup.so libzkgroup_linux_amd64.so
+else ifeq ($(UNAME_HARDWARE_PLATTFORM), aarch64)
+	@echo "get zkgroup $(PLATFORM)"
+	go get -d github.com/nanu-c/zkgroup
+	&& git submodule update \
+	&& cd lib/zkgroup \
+	&& cargo build --release --verbose
+	mv libzkgroup.so libzkgroup_linux_arm64.so
+else
+	@echo architecture not supported
+	exit 1
+endif
+else
+	@echo "platform not supported $(UNAME_S)"
+	exit 1
+endif
+
+copy-zkgroup:
+ifdef GO_VERSION
+	@echo "Found go with version $(GO_VERSION)"
+else
+	@echo go not found, please install go
+	exit 1
+endif
+ifeq ($(UNAME_S), Linux)
+ifeq ($(UNAME_HARDWARE_PLATTFORM), x86_64)
+	@echo "get zkgroup $(PLATFORM)"
+	go get -d github.com/nanu-c/zkgroup
+else ifeq ($(UNAME_HARDWARE_PLATTFORM), aarch64)
+	@echo "get zkgroup $(PLATFORM)"
+	go get -d github.com/nanu-c/zkgroup
+else
+	@echo architecture not supported
+	exit 1
+endif
+else
+	@echo "platform not supported $(UNAME_S)"
+	exit 1
+endif
+
+install-zkgroup:
+ifeq ($(UNAME_S), Linux)
+ifeq ($(UNAME_HARDWARE_PLATTFORM), x86_64)
+	@echo "install libzkgroup to /usr/lib"
+	cp ./libzkgroup_linux_amd64.so /usr/lib/
+else ifeq ($(UNAME_HARDWARE_PLATTFORM), aarch64)
+	@echo "install libzkgroup to /usr/lib"
+	cp ./libzkgroup_linux_arm64.so  /usr/lib/
+
+else
+	@echo architecture not supported
```

Comment:
Idea: what about adding the current architecture to the error message? This is valid for all messages of this type.

`@echo architecture not supported $(UNAME_HARDWARE_PLATTFORM)`